src/frontend/js/modules/previous-page.js

```diff
@@ -1,26 +1,23 @@
-(function(angular) {
-  'use strict';
-
-  angular.module('esn.previous-page', [])
+angular.module('esn.previous-page', [])
 
   .run(function(esnPreviousPage) {
     esnPreviousPage.init();
   })
 
-  .directive('esnBackButton', function(esnPreviousPage) {
+  .directive('esnBackButton', function(esnPreviousPage, $state) {
     return {
       restrict: 'A',
-      link: function(scope, element, attrs) {
+      link: function(scope, element) {
+        // TODO: Write tests for the newly changed logic https://github.com/OpenPaaS-Suite/esn-frontend-calendar/issues/12
+        const defaultState = $state.get().find(state => state.default).name;
```

Comment:
This means we'll have to manually add `default: true` to the default state in each SPA (if the SPA is using `esn-back-button`). For now, only the `esn-frontend-calendar` SPA uses `esn-back-button`, so that's the only place we're going to add it.

Reply:
Should we have the default one. `home` for example.

Reply:
I tweaked it a little bit to also allow the default home state to work (if there's one). Please check the first commit.